sdk/storage/azblob/zt_examples_test.go

```diff
@@ -124,15 +118,325 @@ func Example() {
+	// Set up file to upload
+	fileSize := 8 * 1024 * 1024
+	fileName := "test_upload_file.txt"
+	fileData := make([]byte, fileSize)
+	err := os.WriteFile(fileName, fileData, 0666)
+	handleError(err)
+
+	// Open the file to upload
+	fileHandler, err := os.Open(fileName)
+	handleError(err)
+
+	// close the file after it is no longer required.
+	defer func(file *os.File) {
+		err = file.Close()
+		handleError(err)
+	}(fileHandler)
+
+	// delete the local file if required.
+	defer func(name string) {
+		err = os.Remove(name)
+		handleError(err)
+	}(fileName)
+
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	// Upload the file to a block blob
+	containerName := "testcontainer"
+	blobName := "virtual/dir/path/" + fileName
+	blockSize := 1024
+	parallelism := 3
+	_, err = serviceClient.UploadFile(context.TODO(), containerName, blobName, fileHandler,
+		&azblob.UploadFileOptions{
+			BlockSize:   int64(blockSize),
+			Parallelism: uint16(parallelism),
+			// If Progress is non-nil, this function is called periodically as bytes are uploaded.
+			Progress: func(bytesTransferred int64) {
+				fmt.Println(bytesTransferred)
+			},
+		})
+	handleError(err)
+}
+
+func Example_client_DownloadFile() {
+	// Set up file to download the blob to
+	destFileName := "test_download_file.txt"
+	destFile, err := os.Create(destFileName)
+	handleError(err)
+	defer func(destFile *os.File) {
+		err = destFile.Close()
+		handleError(err)
+	}(destFile)
+
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	// Perform download
+	containerName := "testcontainer"
+	blobName := "virtual/dir/path/" + destFileName
+	blockSize := 1024
+	parallelism := 3
+	downloadOffset := 0
+	downloadCount := azblob.CountToEnd
+	_, err = serviceClient.DownloadFile(context.TODO(), containerName, blobName, destFile,
+		&blob.DownloadFileOptions{
+			Count:       int64(downloadCount),
+			Offset:      int64(downloadOffset),
+			BlockSize:   int64(blockSize),
+			Parallelism: uint16(parallelism),
+			// If Progress is non-nil, this function is called periodically as bytes are uploaded.
+			Progress: func(bytesTransferred int64) {
+				fmt.Println(bytesTransferred)
+			},
+		})
+
+	// Assert download was successful
+	handleError(err)
+
+	fileSize := 8 * 1024 * 1024 // assuming blob size is 8MB.
+
+	// Assert downloaded data is consistent
```

Comment:
We shouldn't have validation code in the samples. This is not test code; it is sample code. The reader assumes the samples works correctly. If we want test, then let's have _test files.

Again, the samples, must be short and sweet - no extra junk that may confuse a reader.

Reply:
Agreed.